handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -239,34 +239,36 @@ protected void deallocate() {
 
         Lock readerLock = context.ctxLock.readLock();
         readerLock.lock();
-        try {
-            ssl = SSL.newSSL(context.ctx, !context.isClient());
-        } finally {
-            readerLock.unlock();
-        }
-        try {
-            networkBIO = SSL.bioNewByteBuffer(ssl, context.getBioNonApplicationBufferSize());
+        synchronized (this) {
```

Comment:
If you look at the stack trace, its actually the C code that is racing in (the SSL callback).